app/org/sagebionetworks/bridge/services/ConsentService.java

```diff
@@ -294,45 +340,78 @@ private void sendWithdrawEmail(Study study, String externalId, Account account,
         
-        BasicEmailProvider.Builder builder = new BasicEmailProvider.Builder()
-                .withStudy(study)
-                .withEmailTemplate(study.getSignedConsentTemplate())
-                .withBinaryAttachment("consent.pdf", MimeType.PDF, consentPdf.getBytes())
-                .withRecipientEmail(participant.getEmail());
-        addStudyConsentRecipients(study, builder);
-        // If the user doesn't have an email, we won't send anything. Note that as of this refactor, 
-        // we're not verifying the email address. Change this when SMS is added.
-        BasicEmailProvider provider = builder.build();
-        if (!provider.getRecipientEmails().isEmpty()) {
+        if (verifiedEmail) {
+            BasicEmailProvider provider = new BasicEmailProvider.Builder()
+                    .withStudy(study)
+                    .withEmailTemplate(study.getSignedConsentTemplate())
+                    .withBinaryAttachment("consent.pdf", MimeType.PDF, consentPdf.getBytes())
+                    .withRecipientEmail(participant.getEmail()).build();
             sendMailService.sendEmail(provider);
+        } else if (verifiedPhone) {
+            sendConsentViaSMS(study, subpop, participant, consentPdf);
+        } else {
+            throw new BadRequestException("Participant does not have a valid email address or phone number");
+        }
+    }
+    
+    private void sendConsentViaSMS(Study study, Subpopulation subpop, StudyParticipant participant,
+            ConsentPdf consentPdf) {
+        String shortUrl = null;
+        try {
+            String fileName = getSignedConsentUrl();
+            DateTime expiresOn = getDownloadExpiration();
+            s3Helper.writeBytesToS3(USERSIGNED_CONSENTS_BUCKET, fileName, consentPdf.getBytes());
+            URL url = s3Helper.generatePresignedUrl(USERSIGNED_CONSENTS_BUCKET, fileName, expiresOn, HttpMethod.GET);
+            shortUrl = urlShortenerService.shortenUrl(url.toString(), SIGNED_CONSENT_DOWNLOAD_EXPIRE_IN_SECONDS);
+        } catch(IOException e) {
+            throw new BridgeServiceException(e);
         }
+        SmsMessageProvider provider = new SmsMessageProvider.Builder()
+                .withStudy(study)
+                .withPhone(participant.getPhone())
+                .withExpirationPeriod(EXPIRATION_PERIOD_KEY, SIGNED_CONSENT_DOWNLOAD_EXPIRE_IN_SECONDS)
+                .withPromotionType()
+                .withSmsTemplate(study.getSignedConsentSmsTemplate())
+                .withToken(BridgeConstants.CONSENT_URL, shortUrl)
+                .build();
+        notificationsService.sendSmsMessage(provider);
+    }
+    
+    protected String getSignedConsentUrl() {
+        return SecureTokenGenerator.INSTANCE.nextToken() + ".pdf";
+    }
+    
+    protected DateTime getDownloadExpiration() {
+        return DateUtils.getCurrentDateTime().plusSeconds(SIGNED_CONSENT_DOWNLOAD_EXPIRE_IN_SECONDS);
```

Comment:
Given weird things like Daylight Savings time, would it be cleaner to have the expiration in days and use plusDays() instead?

Reply:
I don't have any reason not to do this. I don't know how Amazon measures "seven days" as the limit on the expiration value. Changed.

Reply:
It's more complicated to use days because the expiration label and Redis both use seconds... I don't think people will notice if we're off by an hour because we're not giving an exact date and time at which the file will expire.